server/src/main/java/io/crate/role/Role.java

```diff
@@ -166,6 +184,11 @@ public boolean isUser() {
         return properties.login();
     }
 
+    @NotNull
+    public Map<String, Object> jwtProperties() {
```

Comment:
Using record in https://github.com/crate/crate/pull/15515/commits/040282bbe2fa7c1c7ec8f1524ffbfdfa947b4344

For now `Map<String, String>` would work but if we extend payload with non-string claims at some point it wouldn't works.

Reply:
Sure, but it's not planned or expected to extend this.

Reply:
OK, with Map<String, String> wee can do type safe streaming - I'm removing record then as it's not needed. 

In case it's okay to go with `Map<Stirng, String>` and related to https://github.com/crate/crate/pull/15515#discussion_r1479600492 - I think `parser.map` then would work (not ordered, regular map) but I would do reading similar to `Privilege` to be consistent with other code.

Reply:
as discussed internally, keeping record.